### PR TITLE
forwarder: transaction various improvements

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

We can observe in the agent's logs some missing fields in the formatting:
```text
2018-01-04 11:53:35 CET | INFO | (transaction.go:128 in Process) | successfully posted payload to 'https://6-0-0-app.agent.datadoghq.com/intake/?api_key=*************************xxxxx', the agent will only log transaction success every 20 transactions%!(EXTRA string=Accepted)
```

I removed the extra parameter (body) as it's present on the debug log

Also refactor various nits inside the same file.

### CODE REVIEW

File: pkg/forwarder/transaction.go [hunk 1]
```diff
@@ -93,13 +93,18 @@ func (t *HTTPTransaction) Process(ctx context.Context, client *http.Client) erro
...
 		transactionsExpvar.Add("Errors", 1)
-		return fmt.Errorf("Error while sending transaction, rescheduling it: %s", apiKeyRegExp.ReplaceAllString(err.Error(), apiKeyReplacement))
+		return fmt.Errorf("error while sending transaction, rescheduling it: %s", apiKeyRegExp.ReplaceAllString(err.Error(), apiKeyReplacement))
 	}
 	defer resp.Body.Close()
 
-	body, _ := ioutil.ReadAll(resp.Body)
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Errorf("fail to read the response Body: %s", err)
```
Comment (mfpierre) [inline pkg/forwarder/transaction.go]: s/fail/Fail

File: pkg/forwarder/transaction.go [hunk 2]
```diff
@@ -125,15 +130,16 @@ func (t *HTTPTransaction) Process(ctx context.Context, client *http.Client) erro
 	loggingFrequency := config.Datadog.GetInt64("logging_frequency")
 
 	if successfulTransactions.Value() == 1 {
-		log.Infof("successfully posted payload to '%s', the agent will only log transaction success every 20 transactions", logURL, string(body))
-		log.Debugf("payload: %s", logURL, string(body))
-	} else if successfulTransactions.Value()%loggingFrequency == 0 {
-		log.Infof("successfully posted payload to '%s'", logURL, string(body))
+		log.Infof("successfully posted payload to %q, the agent will only log transaction success every 20 transactions", logURL)
```
Comment (mfpierre) [inline pkg/forwarder/transaction.go]: s/successfully/Successfully

File: pkg/forwarder/transaction.go [hunk 3]
```diff
@@ -125,15 +130,16 @@ func (t *HTTPTransaction) Process(ctx context.Context, client *http.Client) erro
...
+		log.Debugf("url: %q payload: %s", logURL, string(body))
```
Comment (mfpierre) [inline pkg/forwarder/transaction.go]: s/url/Url

File: pkg/forwarder/transaction.go [hunk 4]
```diff
@@ -125,15 +130,16 @@ func (t *HTTPTransaction) Process(ctx context.Context, client *http.Client) erro
...
+		return nil
+	}
+	if successfulTransactions.Value()%loggingFrequency == 0 {
+		log.Infof("successfully posted payload to %q", logURL)
```
Comment (mfpierre) [inline pkg/forwarder/transaction.go]: s/successfully/Successfully

File: pkg/forwarder/transaction.go [hunk 5]
```diff
@@ -125,15 +130,16 @@ func (t *HTTPTransaction) Process(ctx context.Context, client *http.Client) erro
...
 		log.Debugf("payload: %s", logURL, string(body))
-	} else {
-		log.Debugf("successfully posted payload to '%s': %s", logURL, string(body))
+		return nil
 	}
-
+	log.Debugf("successfully posted payload to %q: %s", logURL, string(body))
```
Comment (mfpierre) [inline pkg/forwarder/transaction.go]: s/successfully/Successfully

**Review by mfpierre — DISMISSED** [review]
📝

File: pkg/forwarder/transaction.go [hunk 6]
```diff
@@ -125,15 +130,16 @@ func (t *HTTPTransaction) Process(ctx context.Context, client *http.Client) erro
...
 
 	if successfulTransactions.Value() == 1 {
-		log.Infof("successfully posted payload to '%s', the agent will only log transaction success every 20 transactions", logURL, string(body))
-		log.Debugf("payload: %s", logURL, string(body))
-	} else if successfulTransactions.Value()%loggingFrequency == 0 {
-		log.Infof("successfully posted payload to '%s'", logURL, string(body))
-		log.Debugf("payload: %s", logURL, string(body))
-	} else {
-		log.Debugf("successfully posted payload to '%s': %s", logURL, string(body))
+		log.Infof("Successfully posted payload to %q, the agent will only log transaction success every 20 transactions", logURL)
```
Comment (masci) [inline pkg/forwarder/transaction.go]: nit: let's use `loggingFrequency` instead of `20` in the message

**Review by masci — DISMISSED** [review]
🍰

**Review by mfpierre — APPROVED** [review]

**Pull request merged**